pymongo/asynchronous/monitor.py

```diff
@@ -274,22 +287,32 @@ async def _check_once(self) -> ServerDescription:
         Returns a ServerDescription, or raises an exception.
         """
         address = self._server_description.address
+        sd = self._server_description
+
+        # XXX: "awaited" could be incorrectly set to True in the rare case
+        # the pool checkout closes and recreates a connection.
+        awaited = bool(
+            self._pool.conns and self._stream and sd.is_server_type_known and sd.topology_version
+        )
         if self._publish:
             assert self._listeners is not None
-            sd = self._server_description
-            # XXX: "awaited" could be incorrectly set to True in the rare case
-            # the pool checkout closes and recreates a connection.
-            awaited = bool(
-                self._pool.conns
-                and self._stream
-                and sd.is_server_type_known
-                and sd.topology_version
-            )
             self._listeners.publish_server_heartbeat_started(address, awaited)
 
         if self._cancel_context and self._cancel_context.cancelled:
             await self._reset_connection()
         async with self._pool.checkout() as conn:
+            if _SDAM_LOGGER.isEnabledFor(logging.DEBUG):
```

Comment:
Shouldn't this happen at the same point where publish_server_heartbeat_started is called?

Reply:
In theory yes. However, we need a connection for the required logging information, and checking out the connection before publishing the heartbeat has a lot of side effects for other tests.

Reply:
I would expect driverConnectionId to be None on the initial started event. Does the spec not expect that?

Reply:
No such allowance is made: https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring-logging-and-monitoring.md#server-heartbeat-started-log-message.

Reply:
Hmm this was intended to be fixed in DRIVERS-2677.